VM-bf/abac00s/VM-bf.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+    Interpreter Brainfuck (VM-bf)
+    Autor: Adam Bac (abac00s)
+
+    The MIT License (MIT)
+    Copyright (c) 2016 Adam Bac
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation file
+    (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge,
+    publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do
+    so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+    MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+    FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+    WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+    Sposob uzycia (dla programu skompilowanego do pliku VM-bf.exe):
+    > VM-bf <nazwa pliku do wykonania>
+
+    Jesli potrzeba wiekszej ilosci pamieci operacyjnej programu, mozna zmienic stala MEMSIZE.
+*/
+
+#include <cstdio>
+#include <cstdlib>
+#include <stack>
+#include <map>
+#define MEMSIZE 256
+
+char* mem;
+char* prog;
+int pt;
+std::map <int, int> bm;
+
+void clean() {
+    delete [] mem;
+    delete [] prog;
+}
+
+void load_file(const char* fname) {
+    FILE* fp = fopen(fname, "r");
+    if(fp == NULL) {
+        printf("Error: Couldn't open the file %s - process terminated.\n", fname);
+        fclose(fp);
+        exit(0);
+    }
+    fseek(fp, 0, SEEK_END);
+    size_t len = ftell(fp);
```

Comment:
Btw, przy otwarciu pliku w trybie tekstowym (samo "r" pod Windowsem to tryb tekstowy) sztuczka z SEEK_END+ftell nie działa do końca dobrze, ponieważ ftell nie bierze pod uwagę "kompresji" \r\n do \n - czyli de facto jest mniej danych jest mniej niż by na to wynik ftell wskazywał.